src/App.jsx

```diff
@@ -1,10 +1,169 @@
+
+
+// endpoint for all https://restcountries.com/v3.1/all
 
 function App() {
+    const [countries, setCountries] = useState([]);
+    const [searching, setSearching] = useState([]);
+    const [error, setError] = useState('');
+    const [error2, setError2] = useState('');
+    const [formState, setFormState] = useState('');
+
+    async function fetchCountries() {
+        setError('')
+        try {
+            const result = await
+                axios.get('https://restcountries.com/v3.1/all');
+            console.log(result.data);
+            // Data wordt gesorteerd op population van hoog naar laag
+            setCountries(result.data.sort((a, b) => b.population - a.population));
+
+        } catch (e) {
+            console.error(e);
+            setError('Het ophalen van de data is mislukt. Probeer het opnieuw');
+        }
+    }
+
+    // let land = 'nederland';
+
+    async function findCountries() {
+        setError2('')
+        try {
+            const found = await
+                axios.get(`https://restcountries.com/v3.1/name/${formState}?fullText=true`);
+            console.log(found.data);
+            setSearching(found.data);
+
+
+        } catch (event) {
+            console.error(event);
+            setError2(`${formState} bestaat niet. Probeer het opnieuw`)
+        }
+    }
+
+    function handleChange(event) {
+        setFormState(event.target.value);
+
+    }
+
+    function handleSubmit(event) {
+        event.preventDefault();
+        console.log('Form Data:', formState);
+        findCountries();
+    }
+
+    const countryName = searching.length > 0 ? searching[0].name.common : ' ';
+    const countryCapital = searching.length > 0 ? searching[0].capital : ' ';
+    const countryFlag = searching.length > 0 ? searching[0].flags : ' ';
+    const countryPopulation = searching.length > 0 ? searching[0].population : ' ';
+    const countryRegion = searching.length > 0 ? searching[0].region : ' ';
+    const countryBorders = searching.length > 0 ? searching[0].borders.join(', ') : ' ';
+
+    // {calcPop(countryPopulation)}
+
 
     return (
         <>
-            Maak je applicatie hier!
+            <main>
+                <section>
+                    <header>
+                        <div className="worldMap-container">
+                            <img src={worldMap} alt=""/>
+                        </div>
+                    </header>
+                    <h1>World Regions</h1>
+
+
+                    <article>
+                        <div className="btn-wrapper">
+                            {error && <p className="error">{error}</p>}
+                            <button type="button" onClick={fetchCountries}>Fetch Countries</button>
+                        </div>
+                    </article>
+                    {countries.length > 0 && (
+                        <article className="countries-wrapper">
+                            {countries.map((country, index) => (
```

Comment:
Je had hier nog spreading kunnen toepassen, zodat je niet overal `country.` vóór hoeft te zetten
```suggestion
                            {countries.map(({ name, population, region, flags}, index) => (
```